dashboard/test/lib/expired_deleted_account_purger_test.rb

```diff
@@ -1,13 +1,29 @@
 require 'test_helper'
+require 'account_purger'
 require 'expired_deleted_account_purger'
 require_relative '../../../shared/test/spy_newrelic_agent'
 
+# For purposes of this test, reopen AccountPurger and fake the actual
+# account deletion logic.
+class AccountPurger
+  private def really_purge_data_for_account(user)
+    user.update(purged_at: Time.now)
+  end
+end
+
 class ExpiredDeletedAccountPurgerTest < ActiveSupport::TestCase
   freeze_time
 
   def setup
     CDO.hip_chat_logging = false
     CDO.slack_endpoint = nil
+
+    # No uploads
+    ExpiredDeletedAccountPurger.new \
+      deleted_after: 4.days.ago,
+      deleted_before: 2.days.ago
+    ExpiredDeletedAccountPurger.any_instance.stubs(:upload_activity_log).at_least(0)
+    PurgedAccountLog.any_instance.stubs(:upload).at_least(0)
```

Comment:
You shouldn't need the `at_least(0)` part with `stubs`, only with `expects`. 

From http://gofreerange.com/mocha/docs/#Expectation_matching___invocation_order
> Stubs and expectations are basically the same thing. A stub is just an expectation of zero or more invocations. The Expectation#stubs method is syntactic sugar to make the intent of the test more explicit.